### PR TITLE
Add a way for bulk abstracting prototypes

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

Adds a new file, ignoredPrototypes.yml
Files and directories in this file will be read and have all of their contained prototypes be made abstract.

This is a very simple way for servers to bulk exclude prototypes they don't want (clothes, construction recipes, etc) while creating minimal conflicts by minimizing the amount of places that have to be modified. Instead of getting conflicts for a bunch of different clothing items when you remove them, you can only remove the places they appear and sidestep other issues. This also allows maps to be easily ignored for tests while not having to deal with resolving conflicts that may arise from deleting the file itself.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

no cl no fun

### CODE REVIEW

File: Content.Shared/Entry/EntryPoint.cs [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
 
 namespace Content.Shared.Entry
 {
     public sealed class EntryPoint : GameShared
     {
         [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
         [Dependency] private readonly ITileDefinitionManager _tileDefinitionManager = default!;
+        [Dependency] private readonly IResourceManager _resMan = default!;
+
+        private const string IgnoreFile = "/ignoredPrototypes.yml";
```
Comment (metalgearsloth) [inline Content.Shared/Entry/EntryPoint.cs:25]: ResPath.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Review by DrSmugleaf — APPROVED** [review]
It feels like this should probably support multiple files or at least have the one file be settable through a CVar, but I suppose that is only a problem for downstreams of downstreams.

**Comment by EmoGarbage404** [comment]
If i were to add multi-file support, how would you like it to be done? Just a readonly list in Entry Point?

**Comment by DrSmugleaf** [comment]
> If i were to add multi-file support, how would you like it to be done? Just a readonly list in Entry Point?

Could do any .yml files in a directory like the changelog does

**Comment by EmoGarbage404** [comment]
@DrSmugleaf added and tested multi-file support. Now, any file inside of the IgnoredPrototypes directory will be read, rather than just a single file.

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**